spring-web/src/main/java/org/springframework/http/converter/json/KotlinSerializationJsonHttpMessageConverter.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.http.converter.json;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+import kotlinx.serialization.json.Json;
+import kotlinx.serialization.json.JsonConfiguration;
+import kotlinx.serialization.json.JsonDecodingException;
+import kotlinx.serialization.modules.EmptyModule;
+import kotlinx.serialization.modules.SerialModule;
+import org.jetbrains.annotations.NotNull;
+
+import org.springframework.http.HttpInputMessage;
+import org.springframework.http.HttpOutputMessage;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.AbstractHttpMessageConverter;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.http.converter.HttpMessageNotWritableException;
+import org.springframework.http.converter.KotlinSerializationResolver;
+import org.springframework.lang.Nullable;
+import org.springframework.util.StreamUtils;
+
+/**
+ * Implementation of {@link org.springframework.http.converter.HttpMessageConverter} that can read and write JSON using
+ * <a href="https://github.com/Kotlin/kotlinx.serialization">kotlinx.serialization</a>.
+ *
+ * <p>This converter can be used to bind {@code @Serializable} Kotlin classes. It supports {@code application/json} and
+ * {@code application/*+json} with various character sets, {@code UTF-8} being the default.
+ *
+ * @author Andreas Ahlenstorf
+ */
+public class KotlinSerializationJsonHttpMessageConverter extends AbstractHttpMessageConverter<Object> {
+
+	private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
+
+	private final KotlinSerializationResolver resolver = new KotlinSerializationResolver();
+
+	private final Json json;
+
+	/**
+	 * Construct a new {@code KotlinSerializationJsonHttpMessageConverter} with default configuration.
+	 */
+	public KotlinSerializationJsonHttpMessageConverter() {
+		this(JsonConfiguration.getDefault(), EmptyModule.INSTANCE);
+	}
+
+	/**
+	 * Construct a new {@code KotlinSerializationJsonHttpMessageConverter} with custom configuration.
+	 */
+	public KotlinSerializationJsonHttpMessageConverter(JsonConfiguration jsonConfiguration, SerialModule serialModule) {
```

Comment:
```suggestion
	public KotlinSerializationJsonHttpMessageConverter(Json json) {
```

That way the consumer can construct the Json Instance in whichever way they want it to be. The other constructor would change like that:

```diff
public KotlinSerializationJsonHttpMessageConverter() {
-  this(JsonConfiguration.getDefault(), EmptyModule.INSTANCE);
+  this(new Json(JsonConfiguration.getDefault(), EmptyModule.INSTANCE)
}

Reply:
Done.